Automattic/wp-calypso
# Jetpack Cloud | Add Social upsell when it's not active

> <details>
> <summary>When neither Jetpack or Social is active, and say Boost is active, users get the following error when they click on enable publicize.</summary>
>
> ![Image](https://github.com/user-attachments/assets/bcb3b515-cd8d-48af-b25f-d85898746a3d)
> 
> To get around it, we need to show a pricing welcome page the same way we do in my Jetpack. 
> 
> 
> 
> ![Image](https://github.com/user-attachments/assets/dd3e171e-55a4-4910-99c7-3914488d4b6e)
>
> </details>

## Proposed Changes

* Show a pricing card when there is no paid plan on the site
* Show the social activation card when there is a paid plan
* Fix the social RNA card image to replace Twitter with Instagram
* Fix spacing in the RNA CTA which had no space between text and external icon

## Why are these changes being made?


- We want to ensure that the user is not left confused when they don't have the Social or Jetpack plugin active
- We want to guide them holding their hand to ensure they are not lost in the error messages that are shown when the plugins are not active

## Testing Instructions



* Ensure that the site does not have a social plan active
* Ensure that neither Social nor Jetpack is active on the site
* Ensure that another standalone plugin like Boost is active
* Goto Jetpack Cloud `/jetpack-social/:site`
* Confirm that you see the pricing card for Social
* Confirm that the Twitter icon on the image is replaced by Instagram
* Click on "Add Jetpack Social"
* Confirm that the checkout is successful
* Now after adding a Social plan to the site, goto `/jetpack-social/:site` again
* Confirm that you see the social activation card instead of the pricing card
* Confirm that the Twitter icon on the image is replaced by Instagram
* Confirm that you see the "Get started" CTA that points to plugin installation/activation page on the site.
* For a non-admin author, confirm that the link points to the plugin page on wp.org.
* Install/Activate the plugin but disable Social sharing
* Reload the `/jetpack-social/:site` page
* Confirm that you now see "Enable Social sharing" as the button label and it points to the plugin settings page.
* Confirm that it points to Social admin page when it's active, otherwise it points to Jetpack settings > Sharing page
* Confirm that everything works same as before when Jetpack or Social is active with publicize enabled
* Confirm that everything is same as before for Simple and Atomic sites

## Pre-merge Checklist




- When there is no Social plan on the site
    <img width="1301" alt="Screenshot 2024-07-19 at 4 16 38 PM" src="https://github.com/user-attachments/assets/c9445c93-5867-4e09-b466-57b7d8b6b2e5">
- For admin user, when there is a paid plan but none of the plugins is active. Notice the CTA URL at the bottom
    <img width="1303" alt="Screenshot 2024-07-19 at 4 22 53 PM" src="https://github.com/user-attachments/assets/5fed2167-bd91-4eaf-bc09-fd94fb784245">
- For non-admin user, when there is a paid plan but none of the plugins is active. Notice the CTA URL at the bottom
    <img width="1269" alt="Screenshot 2024-07-19 at 4 32 32 PM" src="https://github.com/user-attachments/assets/c324be12-81dc-45fe-a5a0-b26102e99bb0">
- When the plugin is active but sharing is disabled
    <img width="1300" alt="Screenshot 2024-07-19 at 4 34 11 PM" src="https://github.com/user-attachments/assets/5515a467-d090-447e-8f35-cab40035e072">
- When the plugin is active and sharing is enabled
    <img width="1302" alt="Screenshot 2024-07-19 at 4 36 13 PM" src="https://github.com/user-attachments/assets/93286ec7-27a6-41bd-8291-abd1347585a9">
